tests/common/common.h

```diff
@@ -74,16 +84,22 @@ bool check_return_type_bool(returnT returnVal) {
  * @brief Helper function to check two types are equal.
  */
 template <typename expectedT, typename actualT>
-void check_equal_type(sycl_cts::util::logger& log, actualT actualVal,
-                      std::string logMsg) {
+void check_equal_type(actualT actualVal, std::string logMsg) {
   if (typeid(expectedT) != typeid(actualT)) {
-    FAIL(log, logMsg + "\nGot type -> " +
-                  std::string(typeid(actualT).name()) +
-                  "\nExpected type -> " +
-                  std::string(typeid(expectedT).name()));
+    FAIL(logMsg + "\nGot type -> " + std::string(typeid(actualT).name()) +
+         "\nExpected type -> " + std::string(typeid(expectedT).name()));
```

Comment:
I suspect this could be
```suggestion
    FAIL(logMsg + "\nGot type -> " + typeid(actualT).name() +
         "\nExpected type -> " + typeid(expectedT).name());
```

Reply:
Yes, or even better, `operator<<`. Although as far as I can tell, this function is currently not being used anywhere, so it should probably just be removed (in favor of `CHECK(std::is_same_v<T, U>)`).